src/Blef.GameLogic/PokerHands/TwoPairs.cs

```diff
@@ -20,8 +21,30 @@ public override bool IsOnTable(Table table)
             return firstRankCount >= 2 && secondRankCount >= 2;
         }
 
-        protected override int Value => 100 + GetRankValue(first) + GetRankValue(second);
+        protected override int GenericPokerHand => 3;
 
+        protected override int CompareWithinSameGenericPokerHand(PokerHand otherPokerHand)
+        {
+            var otherTwoPair = otherPokerHand as TwoPairs;
+
+            if (otherTwoPair == null)
+            {
+                throw new InvalidOperationException($"This method can compare only {nameof(PokerHand)} with the same GenericPokerHand value");
+            }
+
+            return CompareWithinSameGenericPokerHand(otherTwoPair);
+        }
 
+        int CompareWithinSameGenericPokerHand(TwoPairs otherTwoPair)
```

Comment:
Napisze po polsku bo mi tak wygodniej a mam malo czasu. Z tego co mi wiadomo wyjatki powinno sie rzucac wtedy gdy chce sie wyjsc z metody w jakims szczegolnym wypadku w taki sposob aby klient jesli chce to aby mogl sie o tym szczegolnej sytuacji dowiedziec. Moznaby zyc i pisac kod nie rzucajac wyjatkow ale wtedy musielibysmy udostepnic w interfejsie naszego modulu, klasy obsluge tych sytuacji w postaci na przyklad wzorca obserwator. Niestety takie podejscie niesie za soba taka wade ze moglibysmy zaciemnic nasz interfejs. Klient zmuszony by byl czesto do urzywania null pattern gdyby jakies szczegolne zachowanie by go nie interesowalo. Tak wiec rzucanie wyjatkow jest dobrem ale nalezy go uzywac z glowa. Jesli produkujemy biblioteke na przyklad do edycji ploki tekstowego to aby nie zaciemniac interfejsu nie wymagamy w konstruktorze obiektu ktory sie zasubskrybuje na zdarzenie gdy plik o podanej sciezce nie istnieje. Zaciemniloby to nam interfejs. ALE tutaj w tym wypadku musze sie nie zgodzic z zasadnoscia rzucania wyjatku. Kod ktory w tej grze piszemy uwazam ze nie wymsga rzucania wyjatku. Mozna tak napisac kod aby nie bylo potrzeby rzutowania co jest innym dodatkowo smrodem. Mozna tak naoisac kod aby nie musiec rzucac tu wyjatku. Dodatkowo ty tutaj rzucasz wyjatek nie z powodu ze system moze kiedys wejsc w jakis nieprzewidywany stan. Ty tutaj rzucasz wyjatek gdy programista urzyje twojego kodu w sposob nieporawny dopuszxzajac do tej sytuacji. Troche mi to zalatuje dziwnym nie ufaniem samemu sobie. Rzuce wyjatek gdy podam zly argument. To jest inna odmiana programowania defensywnego tylko nie spraedzasz czy tu jest null a sprawdzasz czy da sie rzutowac obiekt. Ja jestem za pisaniem kodu w taki sposob aby zasady OOP i silne typowanie pomagalo a nie przeszkadzalo.